tools/clang/unittests/SPIRV/CodeGenSpirvTest.cpp

```diff
@@ -1766,6 +1774,18 @@ TEST_F(FileTest, GeometryShaderEmit) { runFileTest("gs.emit.hlsl"); }
 TEST_F(FileTest, ComputeShaderGroupShared) {
   runFileTest("cs.groupshared.hlsl");
 }
+TEST_F(FileTest, ComputeShaderGroupSharedFunctionParam) {
+  setRelaxLogicalPointer();
```

Comment:
this one is using `setRelaxLogicalPointer` instead of `setBeforeHLSLLegalization`. why?

Reply:
I think it would be best to use before-hlsl-legalization.